crates/symbolicator/src/cache.rs

```diff
@@ -26,6 +26,37 @@ use crate::types::Scope;
 /// yet.
 pub const MALFORMED_MARKER: &[u8] = b"malformed";
 
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum MalformedCause {
+    BadObject,
+    DownloadError,
+    Unknown,
+}
+
+impl From<&[u8]> for MalformedCause {
+    fn from(value: &[u8]) -> Self {
+        if value == b"badobject" {
+            Self::BadObject
+        } else if value == b"downloaderror" {
+            Self::DownloadError
+        } else {
+            Self::Unknown
+        }
+    }
+}
+
+impl From<MalformedCause> for &[u8] {
+    fn from(val: MalformedCause) -> Self {
+        match val {
+            MalformedCause::BadObject => b"badobject",
+            MalformedCause::DownloadError => b"downloaderror",
+            // TODO: This could cause us to erase valid causes in the future if we introduce new
+            // variants. Maybe we should store the unrecognized cause inside of Unknown?
+            MalformedCause::Unknown => &[],
```

Comment:
You could solve this by using `MalformedCause::Other(String)` instead of `MalformedCause::Unknown`?

Reply:
it is done